portfolio.py

```diff
@@ -2,13 +2,40 @@
 Allocation of the asset.
 """
 
+import numpy as np
+from thousandaire.constants import TRADING_INSTRUMENTS
+
 class Portfolio(dict):
     """
     Set the portfolio
     """
-    def __init__(self, date):
+    def __init__(self, date, target=None, np_array=None):
         dict.__init__(self)
         self.date = date
```

Comment:
I start to think that we don't really need to keep `date` in this object.
We can always pack date and Portfolio together by Data.
Do you think there are any benefits of keeping this variable here?

Reply:
(we generate lots of Portfolio objects, so the space efficiency is definitely something we should think about)

Reply:
I think maybe alpha writers would keep daily portfolio themselves to generate next portfolio, they may need date? 
However, they can get the date from data we give, it may not be a trouble to make users pack date & portfolio themmselves?

Reply:
1. As I know,, only few alphas must rely on actual date. Most alphas only care about relative time, such as previous 50 days. In such cases, date is not really important.
2. If an alpha must rely on date, the developer definitely knows how to play with datetime. In that cases, it's still not a problem for them to develop the alpha.

That's why I don't really think they need such feature.